target/targetsvc/target_service.C

```diff
@@ -67,4 +68,14 @@ TargetPtr TargetService::getTopLevelTarget() const
     }
     return _targetMap->getTopLevelTarget();
 }
+
+void TargetService::setHwAccessMethod(TargetPtr target, HwAccessMethod accessMethod)
+{
+    target->setAttr<ATTR_HW_ACCESS_METHOD>(accessMethod);
+
+    uintptr_t accessPtr =
+            reinterpret_cast<uintptr_t>(hwaccess::HwAccessIntf::getHwAccessPtr(accessMethod));
+
+    target->setAttr<ATTR_HW_ACCESS_PTR>(static_cast<HwAccessPtrType>(accessPtr));
```

Comment:
Where do we handle the exception that is thrown if the setAttr and getAttr failed? Is the caller expected to handle it? Should that be added to the function header?